psi4/src/psi4/libmints/molecule.cc

```diff
@@ -321,6 +321,10 @@ void Molecule::add_unsettled_atom(double Z, std::vector<std::string> anchor, std
     } else {
         throw PSIEXCEPTION("Illegal geometry specification (neither Cartesian nor Z-Matrix)");
     }
+
+    if ((label != "X") && (label != "x")) {
```

Comment:
did you hit this directly? I really thought `label` should be properly cased by this point.

Reply:
I didn't check the label casing. I noticed that `atoms_` wasn't populated after an `add_unsettled_atoms` call and just copied https://github.com/psi4/psi4/blob/68af15a18c9abf8e1b803c38b1e8d2c89408eb23/psi4/src/psi4/libmints/molecule.cc#L243-L245 from `add_atoms`. I can check if proper casing is enforced by this point and edit the relevant lines from both `add_atoms` and `add_unsettled_atoms` if it is.

Reply:
I think you're in the right